adam/visualization/positioning.py

```diff
@@ -20,50 +20,62 @@
 
 # see https://github.com/pytorch/pytorch/issues/24807 re: pylint issue
 ORIGIN = torch.zeros(3, dtype=torch.float)  # pylint: disable=not-callable
-COLLISION_PENALTY = 10
+GRAVITY_PENALTY = torch.tensor([1], dtype=torch.float)  # pylint: disable=not-callable
+BELOW_GROUND_PENALTY = 2 * GRAVITY_PENALTY
+COLLISION_PENALTY = 5 * GRAVITY_PENALTY
 
 
 @attrs(frozen=True, auto_attribs=True)
 class AdamObject:
     """Used for testing purposes, to attach a name to a bounding box"""
 
     name: str
+    initial_position: Optional[Tuple[float, float, float]]
 
 
-def main() -> None:
-    ball = AdamObject(name="ball")
-    box = AdamObject(name="box")
+def run_model(
+    objs: List[AdamObject], num_iterations: int = 200, yield_steps: Optional[int] = None
+) -> Generator[List[torch.Tensor], None, List[torch.Tensor]]:
+    """
+    Construct a positioning model given a list of objects to position, return their position values.
+    Args:
+        objs: list of AdamObjects requested to be positioned
+        num_iterations: total number of SGD iterations.
```

Comment:
```suggestion
        *num_iterations*: total number of SGD iterations.
```

(and similarly elsewhere)